hllplus/sparse.go

```diff
@@ -124,6 +152,33 @@ func (s *sparseState) encode(hash uint64) uint32 {
 	return s.encodedFlag | normPos<<sparseRhoWBits | uint32(rho)
 }
 
+// TODO: should we squash decodeNormalXXX into single func with 2 retvals?
```

Comment:
yes, we should, much like `encode` we should have a single `decode`, the reason why there are two functions for it in Java is that it cannot do multiple return values